sdks/go/pkg/beam/core/runtime/exec/pardo.go

```diff
@@ -134,40 +134,40 @@ func (n *ParDo) processMainInput(mainIn *MainInput) error {
 	// If the function observes windows, we must invoke it for each window. The expected fast path
 	// is that either there is a single window or the function doesn't observes windows.
```

Comment:
Consider amending the comment here making the optimization explicit.
...doesn't observe windows meaning they can be computed simultaneously.

Reply:
Done, added some comments making this optimization more explicit.